linux_os/guide/system/accounts/accounts-pam/set_password_hashing_algorithm/set_password_hashing_yescrypt_cost_factor_logindefs/rule.yml

```diff
@@ -0,0 +1,59 @@
+documentation_complete: true
+
+title: 'Set yescrypt Cost Factor in /etc/login.defs'
+
+description: |-
+    In <tt>/etc/login.defs</tt>, ensure <tt>YESCRYPT_COST_FACTOR</tt> and
+    <tt>YESCRYPT_COST_FACTOR</tt> has the minimum value of <tt>{{{ xccdf_value("var_password_yescrypt_cost_factor_login_defs") }}}</tt>.
+    For example:
+    <pre>YESCRYPT_COST_FACTOR {{{ xccdf_value("var_password_yescrypt_cost_factor_login_defs") }}}
+    YESCRYPT_COST_FACTOR {{{ xccdf_value("var_password_yescrypt_cost_factor_login_defs") }}}</pre>
+    Notice that if neither are set, they already have the default value of 5.
+    If either is set, they must have the minimum value of {{{ xccdf_value("var_password_yescrypt_cost_factor_login_defs") }}}.
+
+rationale: |-
+    Passwords need to be protected at all times, and hashing is the standard
+    method for protecting passwords. If passwords are not encrypted, they can
+    be plainly read (i.e., clear text) and easily compromised. Passwords
+    that are encrypted with a weak algorithm are no more protected than if
+    they are kept in plain text.
+    <br /><br />
+    Using more hashing rounds makes password cracking attacks more difficult.
+
+severity: medium
+
+identifiers:
+    cce@rhel10: CCE-86594-9
+
+references:
+    nist: IA-5(1)(c),IA-5(1).1(v),IA-7,IA-7.1
+    srg: SRG-OS-000073-GPOS-00041,SRG-OS-000120-GPOS-00061
+
+ocil_clause: 'the cost factor is too low'
+
+ocil: |-
+    Inspect <tt>/etc/login.defs</tt> and ensure that if either
+    <tt>YESCRYPT_COST_FACTOR</tt> or <tt>SHA_CRYPT_MAX_ROUNDS</tt>
```

Comment:
Here we check 2 different keys but the templated OVAL checks only the first one.